plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -1562,6 +1562,303 @@ var _ = Describe("validator", func() {
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeForbiddenError())
+					Expect(err.Error()).To(ContainSubstring("total reserved CPU (kubeReserved + systemReserved) cannot be more than the Node's allocatable CPU"))
+				})
+
+				It("should not allow creation of Shoot if sum of kubeReserved and systemReserved CPU is more than allocatable CPU", func() {
+					worker.Kubernetes.Kubelet.KubeReserved.CPU = &resourceCPU2
+					worker.Kubernetes.Kubelet.SystemReserved.CPU = &resourceCPU2
+
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeForbiddenError())
+					Expect(err.Error()).To(ContainSubstring("total reserved CPU (kubeReserved + systemReserved) cannot be more than the Node's allocatable CPU"))
+				})
+
+				It("should not allow creation of Shoot if reserved memory in the global kubeletConfig is more than allocatable memory and worker kubeletConfig is nil", func() {
+					kubeletConfig.KubeReserved.Memory = &resourceMemory2
+					kubeletConfig.SystemReserved.Memory = &resourceMemory2
+					shoot.Spec.Kubernetes.Kubelet = kubeletConfig
+
+					worker.Kubernetes.Kubelet = nil
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeForbiddenError())
+					Expect(err.Error()).To(ContainSubstring("total reserved memory (kubeReserved + systemReserved) cannot be more than the Node's allocatable memory"))
+				})
+
+				It("should allow creation of Shoot if reserved memory in the global kubeletConfig is more than allocatable memory but the worker kubeletConfig have lesser reserved memory", func() {
+					kubeletConfig.KubeReserved.Memory = &resourceMemory2
+					kubeletConfig.SystemReserved.Memory = &resourceMemory2
+					shoot.Spec.Kubernetes.Kubelet = kubeletConfig
+
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
```

Comment:
Any reason to not use

```suggestion
					Expect(err).ToNot(HaveOccurred())
```

?